sdks/python/apache_beam/transforms/core.py

```diff
@@ -950,6 +957,10 @@ def to_runner_api_parameter(self, context):
                 spec=beam_runner_api_pb2.FunctionSpec(
                     urn=python_urns.PICKLED_DOFN_INFO,
                     payload=picked_pardo_fn_data)),
+            state_specs={spec.name: spec.to_runner_api(context)
+                         for spec in state_specs},
+            timer_specs={spec.name: spec.to_runner_api(context)
+                         for spec in timer_specs},
```

Comment:
Will we eventually use the "self-loop" PCollections as in https://github.com/apache/beam/pull/5883?

(CC: @lukecwik)

Reply:
Eventually. I've had some more thoughts on that that I'v been meaning to write up.